auth/couchdb_session_authenticator.go

```diff
@@ -155,49 +160,59 @@ func (a *CouchDbSessionAuthenticator) Authenticate(request *http.Request) error
 // getCookie returns an AuthSession cookie to be used in a request.
 // A new cookie will be fetched from the session end-point when needed.
 func (a *CouchDbSessionAuthenticator) getCookie() (*http.Cookie, error) {
+	a.mu.Lock()
+	defer a.mu.Unlock()
+
+	a.flushRefreshChannel()
+
 	if a.session == nil || !a.session.isValid() {
-		err := a.syncRequestSession()
+		newSession, err := a.requestSession()
 		if err != nil {
 			return nil, err
 		}
+		a.session = newSession
 	} else if a.session.needsRefresh() {
-		ch := make(chan error)
+		// start a background process to refresh session cookie
+		// if session returned fast enough it'll be updated at the select below
+		// otherwise it'll be updated in next request at flushRefreshChannel
+		ch := make(chan error, 1)
 		go func() {
-			ch <- a.requestSession()
+			session, err := a.requestSession()
+			if err != nil {
+				ch <- err
```

Comment:
If refresh fails groutine places error on `ch` channel and quits. Then error pulled in `select` below.

Reply:
for a record - we discussed it offline and decided that it'd be cleaner just to ignore errors in the background refresh all together, since they do not interrupt main service function.